selectel/data_source_selectel_mks_kube_options.go

```diff
@@ -0,0 +1,237 @@
+package selectel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/selectel/mks-go/pkg/v1/kubeoptions"
+)
+
+func dataSourceMKSFeatureGatesV1() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMKSFeatureGateTypeV1Read,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ru1Region,
+					ru2Region,
+					ru3Region,
+					ru7Region,
+					ru8Region,
+					ru9Region,
+				}, false),
+			},
+			"filter": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kube_version": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"feature_gates": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kube_version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"names": {
```

Comment:
The `Elem` field must be set, otherwise it cause an internal error when trying to run the provider plugin.

The same error occurs for admission controllers.

Please look at you implementation at selectel/resource_selectel_mks_cluster_v1.go#134